### DISCUSSION TITLE
is defaultSettings.yaml for latexindent working?

### DISCUSSION THREAD

**Discussion by NeoFantom** [James-Yu/LaTeX-Workshop#3113] [Q&A]
I tried changing some settings in `defaultSettings.yaml` for latexindent, but it didn't show any effect. I'm not sure if I did things right. But I wanna know, is this file working for latexindent when I format tex file through latex-workshop extension?

**Comment by NeoFantom** [comment]
E.g. when I change `defaultIndent` setting, it has no effect. But changing indent on vscode bottom toolbar works. I'm confused.

**Reply by simphotonics** [reply]
@NeoFantom  some `latexindent` settings do not take effect unless you use the [-m switch](https://latexindentpl.readthedocs.io/en/latest/sec-the-m-switch.htmll). 

For example, to set the maximum number of characters per line to 80 you would have to modify the entry `textWrapOptions` in `defaultSettings.yaml`:
```Perl
textWrapOptions:
        columns: 80
```
and call `latexindent` using:

```Console
$ latexindent -m sample.tex
``` 

---
To have this working in **vscode** you have to add the `-m` switch in Settings > Latex-workshop > Latexindent: Args. 

Alternatively, you can add the following to you settings.json file:
```Json
"latex-workshop.latexindent.args": [
        "-c",
        "%DIR%/",
        "%TMPFILE%",
        "-y=defaultIndent: '%INDENT%'",
        "-m"
    ]
```

**Comment by tamuratak** [comment]
- Fill out all the information required in [the issue template](https://github.com/James-Yu/LaTeX-Workshop/blob/master/.github/ISSUE_TEMPLATE/bug-report.md). 
- Please paste the whole log messages here, not parts of ones. It should start with `Initializing LaTeX Workshop`. It is very important to identify problems.

![スクリーンショット 2020-02-14 20 41 25](https://user-images.githubusercontent.com/10665499/74528024-87b2ef80-4f6a-11ea-9c5d-4f9feff25524.png)

When all of the above is done, we will reopen this issue if needed.
